### PR TITLE
Added Balanced Rearm Ability

### PR DESCRIPTION
**Pull request by Myrl**
Added a new rearm ability that is less OP. It now reduces all cooldowns by 30 seconds instead of refreshing them entirely.

### CODE REVIEW

**Comment by ash47** [comment]
Why would you create a new rearm ability, only to add troll combos that go with it?

Also, conflicts, likely because I deleted the tabs section of abilities.kv

File: src/scripts/abilities/tinker_rearm_lod.lua [hunk 1]
```diff
@@ -0,0 +1,49 @@
+--[[
+	Author: kritth
```
Comment (ash47) [inline src/scripts/abilities/tinker_rearm_lod.lua]: Please remove credits from the file. Credits can be gained via the commit log or the blame section of GitHub.

File: src/scripts/kv/bans.kv [hunk 2]
```diff
@@ -83,6 +83,32 @@
             "zuus_thundergods_vengeance"        "1"
             "zuus_thundergods_wrath"            "1"
         }
+            
+        "tinker_rearm_lod" {
```
Comment (ash47) [inline src/scripts/kv/bans.kv]: Should we really built a skill that has natural troll combos allowed?

File: src/scripts/kv/abilities.kv [hunk 3]
```diff
@@ -858,7 +860,7 @@
                   "140"   "tinker_laser"
                   "141"   "tinker_heat_seeking_missile"
                   "142"   "tinker_march_of_the_machines"
-                  "143"   "tinker_rearm"
+                  "143"   "tinker_rearm||tinker_rearm_lod"
```
Comment (ash47) [inline src/scripts/kv/abilities.kv]: The "tabs" section of abilities.kv has been removed.

**Comment by ash47** [comment]
Just rewrote abilities.kv, it's more stream lined now, but it also created more conflicts

**Comment by Myrl** [comment]
Conflicts in what sense?

**Comment by ash47** [comment]
The master branch has been changed such that git doesn't know how to actually merge your changes as we have both modified the same part of the code.

**Comment by Myrl** [comment]
@ash47 I'll push master to this and push the resolved version. Also, what should I do with the troll combos? Remove them for now and wait for reports?

**Comment by ash47** [comment]
I don't know, that's up for debate, for example, you could add a well
defined rule for which abilities it won't work for, and add that to the
description, for example "doesn't work on ultimate or global abilities"and
then just don't activate it for those

**Comment by Myrl** [comment]
I gave it some thought. This effectively decreases all affected abilities to 60/30/15 seconds cooldown. There's not many normal skills with that long cooldown. I kind of like the "removes n seconds from the cooldown" suggested by @darklordabc if it comes to preventing spamming. I think it's better to add more restrictions to the skill rather than restricting the combinations.

**Comment by ash47** [comment]
https://github.com/ash47/LegendsOfDota/issues/314

**Comment by darklordabc** [comment]
I think this should probably be added to the custom abilities tab, not be grouped with tinker's abilities, like aftershock_lod is. Prehaps we should consider moving aftershock_lod to custom abilities too.

**Comment by darklordabc** [comment]
Any feedback on this ability? If you think its OP, we can change it to 20 second cool-down reduction instead.

**Comment by ash47** [comment]
@jexah

**Comment by ash47** [comment]
I'm still not sold on the idea, I will try it out ingame, if I like it, I will perform a code review. No point reviewing something I don't like.

**Pull request closed without merging**